### ISSUE TITLE
Taint Analysis: the `ResponseFactory` attachment exemption from #1348 clears 1 of 20 real call sites

### ISSUE THREAD

**Issue by alies-dev** [psalm/laravel-psalm-plugin#1416]
**Describe the bug**

Follow-up to #1345, fixed by #1348. `ResponseFactoryTaintHandler` works, but its proof is narrow enough that it exempts almost none of the false-positive class #1345 described. On the Laravel 13 codebase that produced #1345 it clears **1 of 20** `response()->make()` call sites.

`provesLiteralAttachment()` requires all of:

1. exactly 3 arguments, the third positional;
2. the third argument is an `Array_` **at the call site**;
3. every key and value in it is a literal `String_`;
4. exactly one canonical `Content-Disposition` whose value matches `^attachment(?:\s*;\s*\S.*)?$`.

Real export controllers miss on (2) and (3) far more often than they hit. Condition (4) also dropped the `Content-Type` half of #1345's Option 1, so an XML or CSV response that declares only `Content-Type` keeps the sink even though the headers are a fully literal array.

**Code to reproduce**

Each of the four shapes below is a real controller reduced to a documented source. All four report `TaintedHtml`; none is HTML.

```php
// 1. Headers held in a variable. The array is literal, one statement up.
$headers = [
    'Content-Type' => 'text/csv',
    'Content-Disposition' => 'attachment; filename="members.csv"',
];

return response()->make($output, 200, $headers);   // TaintedHtml

// 2. Literal array, interpolated filename. The `attachment` token and its `;` are literal.
return response()->make($output, 200, [
    'Content-Type' => 'text/csv',
    'Content-Disposition' => "attachment; filename=\"{$date}.csv\"",
]);                                                 // TaintedHtml

// 3. Literal array, no Content-Disposition. A sitemap is served inline, on purpose.
return response()->make($xml, 200, ['Content-type' => 'application/xml']);   // TaintedHtml

// 4. The sink is the Response constructor, which the handler does not match at all.
return new \Illuminate\Http\Response($csv, 200, [
    'Content-Type' => 'text/csv',
    'Content-Disposition' => "attachment; filename=\"{$title}.csv\"",
]);                                                 // TaintedHtml
```

**Impacted Versions**

```text
laravel/framework        v13.26.1
psalm/plugin-laravel     dev-master (3c49de52ebf5)
vimeo/psalm              7.0.0-beta19
```

**Psalm config**

```xml
<!-- Default Psalm 7 taint behaviour, no taint-specific overrides. -->
<psalm errorLevel="1" findUnusedBaselineEntry="false" errorBaseline="psalm-baseline.xml">
    <plugins>
        <pluginClass class="Psalm\LaravelPlugin\Plugin">
            <experimental value="true"/>
        </pluginClass>
    </plugins>
</psalm>
```

`TaintedHtml` is not suppressed and the baseline holds no `Tainted*` entry, so every finding below is live.

**Additional context**

### Measured coverage

Every `response()->make()` in `app/`, classified by parsing each file with `nikic/php-parser` and applying the handler's own conditions:

| call shape | sites | passes the proof |
|---|---|---|
| third argument is a `Variable` (3 CSV exports) | 3 | no, fails (2) |
| literal array, `Content-Disposition` is an `InterpolatedString` | 1 | no, fails (3) |
| literal array, `Content-Type` only, no disposition (2 sitemaps, 1 SAML metadata) | 3 | no, fails (4) |
| 2 arguments plus a chained `->header('Content-Type', ...)` (7 certificate images, 1 XML webhook reply) | 8 | no, fails (1) |
| literal array with a literal `attachment` disposition | 1 | **yes** |
| genuinely HTML, no headers | 1 | n/a, correctly kept |

Shape 4 in the repro is a second sink entirely: `stubs/common/Http/Response.phpstub:9` marks `Illuminate\Http\Response::__construct`'s `$content` as an `html` sink, and `contentLocationOfMakeSink()` only recognises the three `::make` journey labels. In this codebase that constructor is reached through a shared `CsvReportPresenter::download()` that every CSV export funnels into, so it is the single highest-value site to exempt and currently has no path to it at all.

### Proposed widenings

Ranked by sites recovered. Each keeps the "fail toward a retained finding" property.

1. **Accept a variable whose array literal is visible in the same scope.** The handler already re-parses the sink's file and walks its statements, so a local const-fold is within reach: resolve `$headers` when the enclosing function-like assigns it exactly one array literal and never reassigns or mutates it (`$headers[...] =`, `array_merge`, passing it by reference all disqualify). Anything unresolvable keeps the sink. Recovers 3 sites here.

2. **Accept an `InterpolatedString` or `Concat` disposition whose literal prefix already proves `attachment`.** Test the first part: it must be an `InterpolatedStringPart` (or a `String_` on the left of a `Concat`) whose value matches `^attachment\s*;`. Interpolation then begins strictly after the parameter separator, and nothing in a parameter list can retract the token, so the proof is as strong as the all-literal one. `filename="{$name}.csv"` is the dominant real spelling of this header. Recovers 1 site here and, combined with (1), the shared presenter in shape 4.

3. **Cover `Illuminate\Http\Response::__construct`.** Same sink kind, same proof, one more journey label plus a `New_` branch in `findMakeCallWithContentAt()`. The constructor's third parameter is already `array $headers`, so the argument-shape check is unchanged.

4. **Restore the `Content-Type` proof from #1345's Option 1.** A literal `Content-Type` naming a type browsers never render as HTML (`application/xml`, `text/csv`, `text/calendar`, `application/pdf`, `image/*`, ...) is the same argument #1318 already accepted for `json()`, and strictly stronger than it: `json()` was exempted on an implied content type, this one is written at the call site. Note the fold in shape 3 above is `Content-type`, which `HeaderBag` normalises, so this check needs the same `strtr()` folding the disposition check uses. Recovers 3 sites here.

5. Optional, lowest value: a 2-argument `make()` followed by a chained `->header('Content-Type', ...)`. Reaching it means walking to the `MethodCall` parent of the matched node, which the current "find the call whose content argument spans these bounds" search does not give you for free. 8 sites here, but all 8 currently serve rendered images and reply bodies whose content is not tainted, so this is speculative until one of them takes user data.

(1) and (2) together are what the shape in #1345 actually needed. If only one lands, (2) is the cheaper of the pair and (1) the one that unblocks more sites.

### Out of scope

`make()`'s `$headers` is not modelled as a `header` sink. Shape 2 above interpolates an unescaped, user-controlled team name straight into a quoted `filename`, and Psalm says nothing. That is a coverage gap rather than a false positive, so it belongs in its own issue.
